.github/workflows/node.js.yml

```diff
@@ -6,19 +6,64 @@ name: Node.js CI
 on: [push, pull_request]
 
 jobs:
-  build:
-
+  common:
+    name: Common checks (Node.js v${{ matrix.node-version }})
     runs-on: macos-latest
-
     strategy:
       matrix:
         node-version: [12.x, 14.x, 16.x]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - uses: bahmutov/npm-install@v1
+        with:
+          useRollingCache: true
+      - run: npm run test:common
 
+  ios-simulator:
+    name: ios-simulator checks (Node.js v${{ matrix.node-version }} / iOS v${{ matrix.ios-version }} / Xcode v${{ matrix.xcode-version }} / ${{ matrix.device-name }})
+    runs-on: macos-latest
+    strategy:
+      matrix:
+        device-name: [iPhone 12, iPhone 11]
+        include:
+          - device-name: iPhone 12
+            xcode-version: 12.4
+            ios-version: 14.4
+          - device-name: iPhone 12
+            xcode-version: 12.3
+            ios-version: 14.3       
+          - device-name: iPhone 11
+            xcode-version: 11.7
+            ios-version: 13.7
     steps:
-    - uses: actions/checkout@v2
-    - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v1
-      with:
-        node-version: ${{ matrix.node-version }}
-    - run: npm install
-    - run: npm test
+      - uses: actions/checkout@v2
+      - uses: bahmutov/npm-install@v1
+        with:
+          useRollingCache: true
+      - name: Configure Homebrew cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/Library/Caches/Homebrew/applesimutils--*
+            ~/Library/Caches/Homebrew/downloads/*--applesimutils-*
+          key: brew-${{ hashFiles('.github/brew-formulae') }}
+          restore-keys: brew-
+      - name: Update Homebrew
+        run: |
+          brew tap wix/brew
+          brew update --preinstall
+          cat "$(brew --repository)/Library/Taps/wix/homebrew-brew/applesimutils.rb" > .github/brew-formulae
+      - name: Install Homebrew dependencies
+        env:
+          HOMEBREW_NO_AUTO_UPDATE: 1
+        run: brew install applesimutils
+      - name: Tests (Node.js v${{ matrix.node-version }} / iOS v${{ matrix.ios-version }} / Xcode v${{ matrix.xcode-version }} / ${{ matrix.device-name }})
+        run: scripts/test-ios-simulator.sh
```

Comment:
I'm not quite sure if we want to combine unit and integration tests. I usually split them into stages, because it does not make sense to run e2e stage if unit tests fail. Also, for unit tests it makes more sense to try different node version and single xcode version while for e2e it's vice-versa (latest node and multiple Xcodes)

Reply:
OK, I will split them up.